### PR TITLE
Delete folders when empty for localfs

### PR DESCRIPTION
**Pull request by erinpentecost**
https://github.com/lytics/lio/pull/24601

### CODE REVIEW

File: localfs/store.go [hunk 1]
```diff
@@ -278,11 +279,32 @@ func (l *LocalStore) Get(ctx context.Context, o string) (cloudstorage.Object, er
 // Delete the object from underlying store.
 func (l *LocalStore) Delete(ctx context.Context, obj string) error {
 	fo := path.Join(l.storepath, obj)
-	os.Remove(fo)
+	if err := os.Remove(fo); err != nil {
+		return err
```
Comment (mattayes) [inline localfs/store.go:283]: [NIT] Should we start wrapping these errors?

```suggestion
		return fmt.Errorf("removing obj: %w", err)
```

Same below.

File: localfs/store.go [hunk 2]
```diff
@@ -278,11 +279,32 @@ func (l *LocalStore) Get(ctx context.Context, o string) (cloudstorage.Object, er
...
+	}
+	if _, err = dir.Readdirnames(1); errors.Is(err, io.EOF) {
+		dir.Close()
+		// it's empty, so remove it.
+		if err := os.Remove(l.storepath); err != nil {
+			return err
+		}
+	} else {
+		dir.Close()
+	}
```
Comment (mattayes) [inline localfs/store.go:306]: [NIT] `os.ReadDir()` does this for us:

```suggestion
	dirs, err := os.ReadDir(l.storepath)
	if err != nil {
		return fmt.Errorf("reading dir=%s: %w", l.storepath, err)
	}
	if len(dirs) == 0 {
		// it's empty, so remove it.
		if err := os.Remove(l.storepath); err != nil {
			return fmt.Errorf("removing directory: %w", err)
		}
	}
```

Comment (erinpentecost) [reply localfs/store.go:306]: This will iterate on all the docs in the dir, which will be slow for folders with lots of files.

**Review by mattayes — APPROVED** [review]
:shipit:

**Review by sgarciapdx — APPROVED** [review]

**Pull request merged**